evaluate_segmentation.py

```diff
@@ -0,0 +1,179 @@
+
+    # VALIDATION: anticipate problems with imagery and label (if provided) before entering main for loop
+    valid_gpkg_set = set()
+    for info in tqdm(list_img, desc='Validating ground truth'):
+        if not 'gpkg' in info.keys() and not info['gpkg']:
+            raise ValueError(f"No ground truth was inputted to evaluate with")
+        elif not Path(info['gpkg']).is_file():
+            raise FileNotFoundError(f"Couldn't locate ground truth to evaluate with.")
+
+        if info['gpkg'] not in valid_gpkg_set:
+            validate_num_classes(vector_file=info['gpkg'],
+                                 num_classes=num_classes,
+                                 attribute_name=attribute_field,
+                                 ignore_index=dontcare,
+                                 attribute_values=attr_vals)
+            assert_crs_match(info['tif'], info['gpkg'])
+            valid_gpkg_set.add(info['gpkg'])
+
+    logging.info('\nSuccessfully validated label data for benchmarking')
+
+    gdf_ = []
+    gpkg_name_ = []
+
+    for info in tqdm(list_img, desc='Evaluating from input list', position=0, leave=True):
+        local_img = Path(info['tif'])
+        Path.mkdir(working_folder.joinpath(local_img.parent.name), parents=True, exist_ok=True)
+        inference_image = working_folder / local_img.parent.name / f"{local_img.stem}_inference.tif"
+        if not inference_image.is_file():
+            raise FileNotFoundError(f"Couldn't locate inference to evaluate metrics with. Make inferece has been run "
+                                    f"before you run evaluate mode.")
+
+        pred = rasterio.open(inference_image).read()[0, ...]
+
+        local_gpkg = Path(info['gpkg'])
+
+        logging.info(f'\nBurning label as raster: {local_gpkg}')
+        with rasterio.open(local_img, 'r') as raster:
+            local_img = clip_raster_with_gpkg(raster, local_gpkg)
+
+        raster_clipped = rasterio.open(local_img, 'r')
+        logging.info(f'\nReading clipped image: {raster_clipped.name}')
+        inf_meta = raster_clipped.meta
+
+        label = vector_to_raster(vector_file=local_gpkg,
+                                 input_image=raster_clipped,
+                                 out_shape=(inf_meta['height'], inf_meta['width']),
+                                 attribute_name=attribute_field,
+                                 fill=0,  # background value in rasterized vector.
+                                 attribute_values=attr_vals)
+        if debug:
+            logging.debug(f'\nUnique values in loaded label as raster: {np.unique(label)}\n'
+                          f'Shape of label as raster: {label.shape}')
+
+        gdf = metrics_per_tile(label_arr=label, pred_img=pred, input_image=raster_clipped, chunk_size=chunk_size,
+                               gpkg_name=local_gpkg.stem, num_classes=num_classes)
+
+        gdf_.append(gdf.to_crs(4326))
```

Comment:
This is where the features are reprojected to lat/long. This step seems necessary before concatenating all geodataframes at line 172.

Reply:
Yeah, slightly different from what is implemented in inference script( see comment above). Confirm it works as intended